src/APICore.ts

```diff
@@ -17,57 +18,41 @@ function convertToQueryString(parameters: any) {
     return parameters ? `?${new URLSearchParams(Object.entries(removeEmptyEntries(parameters))).toString()}` : '';
 }
 
-export interface APIConfiguration {
-    organizationId: string;
-    accessTokenRetriever: () => string;
-    host?: string;
-    responseHandlers?: IRestResponseHandler[];
-}
-
 export default class API {
     static orgPlaceholder = '{organizationName}';
 
     constructor(private config: APIConfiguration) {}
 
-    async get<T>(url: string, queryParams?: any, args: RequestInit = {method: 'get'}): Promise<IRestResponse<T>> {
+    async get<T>(url: string, queryParams?: any, args: RequestInit = {method: 'get'}): Promise<T> {
         return await this.request<T>(url + convertToQueryString(queryParams), args);
     }
 
     async post<T>(
         url: string,
         body: any,
         args: RequestInit = {method: 'post', body: JSON.stringify(body)}
-    ): Promise<IRestResponse<T>> {
+    ): Promise<T> {
         return await this.request<T>(url, args);
     }
 
-    async put<T>(
-        url: string,
-        body: any,
-        args: RequestInit = {method: 'put', body: JSON.stringify(body)}
-    ): Promise<IRestResponse<T>> {
+    async put<T>(url: string, body: any, args: RequestInit = {method: 'put', body: JSON.stringify(body)}): Promise<T> {
         return await this.request<T>(url, args);
     }
 
-    async delete<T = void>(url: string, args: RequestInit = {method: 'delete'}): Promise<IRestResponse<T>> {
+    async delete<T = void>(url: string, args: RequestInit = {method: 'delete'}): Promise<T> {
         return await this.request<T>(url, args);
     }
 
-    private handleResponse<T>(response: Response) {
-        const responseHandler = this.handlers.filter((handler) => handler.canProcess(response))[0];
-        return responseHandler.process<T>(response);
-    }
-
-    private get handlers(): IRestResponseHandler[] {
+    private get handlers(): ResponseHandler[] {
         const customHandlers = this.config.responseHandlers || [];
-        return [...customHandlers, ...Handlers];
+        return [...customHandlers, ...defaultResponseHandlers];
```

Comment:
I'm wondering what's the best option here. If we concat handler then all of the custom one will be ran before ours but the order is important.

I could see someone wanting to use our handler for the 204 and the success but overrides the one for the errors. In this case I think it would be best to export all of our handlers and here do something like:

```suggestion
        return customHandlers.length ? customHandlers : defaultResponseHandlers;
```

The user could then do something like:

```javascript
{
  responseHandlers: [
    PlatformClient.Handlers.noContent,
    PlatformClient.Handlers.success,
    myHandlers.serverError,
    myHandlers.clientError,
  ],
}
```

Reply:
Your concern makes a lot of sense, I totally agree this is something the users will want to do.

Reply:
done